csharp/src/Ice/LocatorInfo.cs

```diff
@@ -13,26 +13,35 @@ namespace ZeroC.Ice
     /// <summary>The locator info class caches information specific to a given locator proxy. The communicator holds a
     /// locator info instance per locator proxy set either with Ice.Default.Locator or the proxy's Locator property. It
     /// caches the locator registry proxy and keeps track of requests to the locator to prevent multiple concurrent
-    /// locator requests for the same adapter or well-known object. The locator info also holds a reference on a locator
-    /// table that caches endpoints and resolved references for adapters or well-known objects.
-    /// TODO: refactor to support new Locator API (for new protocol/encoding). We could consider merging the locator
-    /// table and locator info classes as well.</summary>
+    /// locator requests for the same adapter or well-known object.</summary>
     internal sealed class LocatorInfo
     {
         internal ILocatorPrx Locator { get; }
-        private readonly bool _background;
+
+        private readonly ConcurrentDictionary<string, (TimeSpan InsertionTime, IReadOnlyList<Endpoint> Endpoints)>
```

Comment:
Could we use the new C# 9.0 target-typed new here?

Reply:
Yes, fixed.